src/braket/circuits/serialization.py

```diff
@@ -39,12 +39,14 @@ class SerializableProgram(ABC):
     def to_ir(
         self,
         ir_type: IRType = IRType.OPENQASM,
+        highlight: bool = False,
```

Comment:
I love this feature! I'm just not sure whether `highlight` makes sense as a top-level API argument. It's never useful in the context of running a program. My vote would be to have a separate helper function for this in a (not-yet-existing) visualizations module. So instead of `program.to_ir(highlight=True)` someone could do `highlight(program.to_ir())` or even just `highlight(program)`. 

As a bonus, that makes this functionality easily available to anyone who has an OpenQASM string (say, generated directly from oqpy or the Circuit class). Although I guess that's already possible in like one line of code by just using pygments directly...

For IPython use cases, we could additionally just make the object returned from `to_ir()` implement `_repr_pretty_` (using pygments TerminalFormatter) and/or `_repr_html_` (using pygments HTMLFormatter). See docs here: https://ipython.readthedocs.io/en/stable/config/integrating.html. Then in Jupyter if a cell returns `program.to_ir()` it would print out as highlighted by default. And if someone does `print(program.to_ir())` they would get the plaintext. I think this is consistent with how I've seen other libraries work. But if anyone knows of counterexamples to that I'm open to those too.